x-pack/plugins/security_solution/public/resolver/view/use_resolver_query_params.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useCallback, useMemo } from 'react';
+// eslint-disable-next-line import/no-nodejs-modules
+import querystring from 'querystring';
+import { useHistory, useLocation } from 'react-router-dom';
+import { CrumbInfo } from './panels/panel_content_utilities';
+
+export function useResolverQueryParams(documentLocation: string) {
```

Comment:
instead of passing this around, put the `documentLocation` in the action dispatched by `useStateSyncingActions`, then use `useSelector` in this hook to get it.

Reply:
Are there some cool things you could do by having it passed in as an argument like this, though? Just thinking in the future stuff like "pre-focusing" a Resolver by pushing to query params (I guess if you knew its document location) before you open it?